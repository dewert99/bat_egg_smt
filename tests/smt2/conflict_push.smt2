(declare-sort S)
(declare-const a S)
(declare-const b S)
(declare-fun f (S) Bool)
(assert (f a))
(assert (not (f b)))
(push)
(assert (= a b))
(push)
(check-sat)
(pop)
(check-sat)
(pop)
(check-sat)
(get-model)
(assert (= a b))
(push)
(check-sat)
(pop)
(check-sat)