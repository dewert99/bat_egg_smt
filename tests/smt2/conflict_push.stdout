unsat
unsat
sat
(
 (define-fun a () S @S_3)
 (define-fun b () S @S_4)
 (define-fun f ((x!0 S)) Bool
  (ite (= x!0 @S_3) true
  (ite (= x!0 @S_4) false
   f!default)))
)
unsat
unsat
